presto-tests/src/test/java/io/prestosql/execution/TestDiagnosisWarning.java

```diff
@@ -0,0 +1,140 @@
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+import static org.testng.Assert.fail;
+
+public class TestDiagnosisWarning
+{
+    private LocalQueryRunner queryRunner;
+
+    @BeforeClass
+    public void setUp()
+    {
+        queryRunner = new LocalQueryRunner(testSessionBuilder()
+                .setCatalog("local")
+                .setSchema("tiny")
+                .build());
+
+        queryRunner.createCatalog(
+                queryRunner.getDefaultSession().getCatalog().get(),
+                new TpchConnectorFactory(1),
+                ImmutableMap.of());
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+    {
+        queryRunner.close();
+    }
+
+    @Test
+    public void testWarningSession()
+    {
+        WarningCode warningCode = StandardWarningCode.INTEGER_DIVISION.toWarningCode();
+        // sessionProperties = null
+        assertPlannerWarnings(queryRunner,
+                "SELECT 1/2",
+                ImmutableMap.of(),
+                Optional.empty());
+
+        // session enable_query_diagnosis_warning = true
+        assertPlannerWarnings(queryRunner,
+                "SELECT 1/2",
+                Optional.of(warningCode));
+    }
+
+    @Test
+    public void testIntegerDivisionWarning()
+    {
+        WarningCode warningCode = StandardWarningCode.INTEGER_DIVISION.toWarningCode();
+        assertPlannerWarnings(queryRunner,
+                "SELECT 1/2",
+                Optional.of(warningCode));
+        assertPlannerWarnings(queryRunner,
+                "SELECT a/2 FROM (VALUES (1)) as t(a)",
+                Optional.of(warningCode));
+        assertPlannerWarnings(queryRunner,
+                "SELECT 1/2.0",
+                Optional.empty());
+    }
+
+    @Test
+    public void testDeprecatedFunction()
+    {
+        WarningCode warningCode = StandardWarningCode.DREPRCATED_FUNCTION.toWarningCode();
+        assertPlannerWarnings(queryRunner,
+                "SELECT 123",
+                Optional.empty());
+        assertPlannerWarnings(queryRunner,
+                "SELECT json_array_get('[1, 2, 3]', 0)",
+                Optional.of(warningCode));
+        assertPlannerWarnings(queryRunner,
+                "SELECT json_array_get(a, 0) FROM (VALUES ('[1, 2, 3]')) AS t(a)",
+                Optional.of(warningCode));
+    }
+
+    private static void assertPlannerWarnings(LocalQueryRunner queryRunner, @Language("SQL") String sql, Optional<WarningCode> expectedWarning)
+    {
+        assertPlannerWarnings(queryRunner, sql, ImmutableMap.of("enable_query_diagnosis_warning", "true"), expectedWarning);
+    }
+
+    private static void assertPlannerWarnings(LocalQueryRunner queryRunner, @Language("SQL") String sql, Map<String, String> sessionProperties, Optional<WarningCode> expectedWarning)
```

Comment:
Can we change `Optional<WarningCode>` to `List<WarningCode>` for multiple warning case?